### ISSUE TITLE
Deleting Characters Causes Analyzer to Read Wrong World Data

### ISSUE THREAD

**Issue by Razzmatazzz** [Razzmatazzz/RemnantSaveManager#15]
If you have multiple characters and delete any character(s) other than the last one, the world analyzer will likely no longer display the correct information for your remaining character(s). If you create new characters to get back up to the same number you had previously, the issue should be fixed. Given the way the game handles characters and world data, it's not really possible to account for this in the analyzer, so the issue wont' be fixed.

**Comment by AndrewSav** [comment]
What is the particular technical challenge here?

**Comment by Razzmatazzz** [comment]
The challenge is that all characters are listed in the profile.sav file and each character's world data is saved in save_X.sav files (with the X being a number, starting at 0 for the first save). Nothing that's human-readable in the profile.sav file connects a character to the world data except for the order in which they appear. So if you have three characters and delete the middle one, I believe you end up with a save_0.sav and a save_2.sav. I'm unsure when/if save_2 becomes renamed to save_1, so either on the deletion of that character or the creation of another, it's possible that the world data files will become "out of order" compared to the world save file names.

**Comment by AndrewSav** [comment]
Right, so this is because we do not know the format and no one reverse engineered this part. That makes sense. Thank you for answering this, and other questions.

**Comment by AndrewSav** [comment]
Not sure what this is, but I just tried deleting the middle save and the others got renamed to cover the hole...

**Comment by Razzmatazzz** [comment]
I've received reports of folks getting incorrect world analyzer information due to the presence of extra save_x.sav files. To fix this, users should log into each of their existing characters to trigger a save, then close Remnant and RemnantSaveManager. Then users should view their save files (%AppData%\Local\Remnant\Saved\SaveGames) and make note of how many save_x.sav files there are. These files generally start with save_0.sav and the number increases by one for each additional character. If extra save files is the issue, there will be more of these save_x.sav files than the user has characters. The "inactive" save files should be those with the oldest "modified" date since the user just triggered a save update on existing characters. These inactive files can be deleted, and then Remnant and RemnantSaveManager can be relaunched and everything should work.
